src/js/components/homepage/visualizations/TreeMap.jsx

```diff
@@ -84,29 +96,90 @@ export default class TreeMap extends React.Component {
                 y1={n.y1}
                 total={n.parent.value}
                 key={i}
-                color={colors[i]} />
+                color={colors[i]}
+                chosen={this.state.category}
+                toggleTooltip={this.toggleTooltip}
+                showOverlay={this.state.showOverlay} />
         );
 
         this.setState({
             finalNodes: nodes
         });
     }
 
+    toggleTooltip(cat, value) {
+        const descSet = this.state.descriptions;
+        // find index of object item on matching cat name
+        let descIndex = '0';
+        if (cat !== 'none') {
+            descIndex = _.findIndex(descSet, { name: cat });
+        }
+
+        // set it to desc value
+        let desc = '';
+        if (cat !== 'none') {
+            desc = descSet[descIndex].value;
+        }
+
+        // set the state
+        this.setState({
+            category: cat,
+            description: desc,
+            individualValue: value,
+            showOverlay: false
+        });
+        const newNodes = [];
+
+        // create a copy of finalNodes
+        // update the copy
+        this.state.finalNodes.forEach((i) => {
+            const originalNode = i;
+            const updatedNode = update(originalNode, {
```

Comment:
I'll be honest - I don't like this. You should never be directly manipulating props and this comes too close to doing that (though I recognize you're actually cloning the component and passing in new props during the clone process). Furthermore, I see no indication that `update` is safe for use with components...I'm pretty sure React components are not meant to be cloned/copied.

The easiest alternative is slightly more computationally expensive, but you would simply add a new if statement to the `componentWillReceiveProps` function to re-calculate the treemap when the overlay/category/etc states change and pass the new props down to the child treemap cells in the `buildTree` function.

A more complex approach would be to hold references to the child cell instances in your state and change their highlight props as necessary individually.